src/Routing/ContentfulSlugMatcher.php

```diff
@@ -218,13 +218,16 @@ private function getMatchingEntry(string $requestUri)
             $cacheHit->set($entry);
 
             if ($cacheTTL = $this->getCacheTTL()) {
-
                 $cacheHit->expiresAfter($cacheTTL);
             }
 
             if (!$this->isCacheIgnored) {
-                if (method_exists($cacheHit, 'tag')) {
-                    $cacheHit->tag($this->getCacheTags($entry));
+                if (method_exists($cacheHit, 'tag') &&
+                    ($slugFieldName = $this->getSlugField()) &&
+                    (isset($entry[$slugFieldName])) &&
+                    ($slugField = $entry[$slugFieldName])
+                ) {
+                    $cacheHit->tag($this->getRoutingCacheId($slugField));
```

Comment:
I would suggest another value:
 The cache reset service does nothing with the routing cache id, but it uses the entry id as a tag. So, you use the id as tag.)